src/v/redpanda/application.h

```diff
@@ -268,25 +305,33 @@ class application {
      * @return the future returned by start()
      */
     template<typename Service, typename... Args>
-    ss::future<> construct_service(ss::sharded<Service>& s, Args&&... args) {
-        _deferred.emplace_back([&s] { s.stop().get(); });
+    ss::future<> construct_service(
+      std::string_view service_name, ss::sharded<Service>& s, Args&&... args) {
+        _deferred.emplace_back([this, &s, name = ss::sstring(service_name)] {
```

Comment:
nit: why the function argument is not ss::sstring instead and `std::move` here used instead? That's a rule of thumb iirc. Give the opportunity for the caller to avoid a copy if possible. Really a nit because it does not matter much here but still wonder why you choose to create a string view, then ss::sstring from it instead.

Reply:
I'll try alternative approach

Reply:
this problem no longer exist